dmscripts/notify_buyers_to_award_closed_briefs.py

```diff
@@ -0,0 +1,139 @@
+
+
+def _filter_brief_users_by_optional_user_id_list(brief, user_id_list):
+    """Filter the brief's users by restricted user ID list if present
+    """
+    if user_id_list:
+        return filter(lambda user: user['id'] in user_id_list, brief['users'])
+    else:
+        return brief['users']
+
+
+def _filter_briefs_by_optional_user_id_list(closed_briefs, user_id_list):
+    """Return briefs if they have any users
+    These users can be optionally filtered by the user_id_list
+    """
+    return [
+        brief for brief in closed_briefs if any(
+            _filter_brief_users_by_optional_user_id_list(brief, user_id_list)
+        )
+    ]
+
+
+def _create_context_for_brief(brief):
+    return {
+        'brief_id': brief['id'],
+        'brief_title': brief['title'],
+        'utm_date': date.today().strftime("%Y%m%d")
+    }
+
+
+def notify_users(notify_client, notify_template_id, brief, user_id_list):
+    failed_users = []
+    if brief['users']:
+        context_data = _create_context_for_brief(brief)
+        brief_users = _filter_brief_users_by_optional_user_id_list(brief, user_id_list)
+
+        for user in brief_users:
+            if user['active']:
+                try:
+                    notify_client.send_email(
+                        user['emailAddress'], notify_template_id, context_data, allow_resend=False
+                    )
+                except EmailError:
+                    failed_users.append(user['id'])
+
+    return failed_users
+
+
+def _check_that_brief_closing_date_before_the_offset(date_closed, offset_days):
+    if date_closed:
+        date_closed = date_helpers.get_date_closed(date_closed)
+        if date_closed > (datetime.utcnow() - timedelta(days=offset_days)).date():
+            return False
+        return date_closed
+    return date.today() - timedelta(days=offset_days)
+
+
+def main(
+    data_api_url, data_api_access_token, notify_api_key, notify_template_id, date_closed=None, dry_run=None,
+    user_id_list=None, offset_days=28
+):
+    """
+    Send emails to buyers via Notify, reminding them to award their closed briefs
+    """
+    logger.info("Data API URL: {data_api_url}", extra={'data_api_url': data_api_url})
+
+    date_closed = _check_that_brief_closing_date_before_the_offset(date_closed, offset_days)
+    if not date_closed:
+        logger.error('Not allowed to notify about briefs that closed less than {} days ago', offset_days)
+        return False
+
+    data_api_client = dmapiclient.DataAPIClient(data_api_url, data_api_access_token)
+    if not dry_run:
+        notify_client = DMNotifyClient(notify_api_key, logger=logger)
+
+    closed_briefs = brief_data_helpers.get_closed_briefs(data_api_client, date_closed)
+    if not closed_briefs:
+        logger.info("No briefs closed on {date_closed}", extra={"date_closed": date_closed})
+        return True
+
+    # If user list supplied, check that the briefs have at least one user in the list
```

Comment:
The use of the filtering briefs stuff in here is hard to understand. I think that is because it is not too clear what is normal functionality and what is functionality only needed in case of a script failure and to be used as a developer manually rerunning something.

It might just be a case of changing the comment to something like (if this is correct):
If user list supplier, restrict/filter briefs to only include those owned by those users